crates/biome_rule_options/src/use_unique_element_ids.rs

```diff
@@ -1,6 +1,45 @@
+use biome_console::markup;
+use biome_deserialize::{
+    DeserializableValidator, DeserializationContext, DeserializationDiagnostic, TextRange,
+};
 use biome_deserialize_macros::Deserializable;
+use rustc_hash::FxHashSet;
 use serde::{Deserialize, Serialize};
+
 #[derive(Default, Clone, Debug, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase", deny_unknown_fields, default)]
-pub struct UseUniqueElementIdsOptions {}
+#[deserializable(with_validator)]
+pub struct UseUniqueElementIdsOptions {
+    /// Component names that accept an `id` prop that does not translate
+    /// to a DOM element id.
+    pub excluded_components: FxHashSet<Box<str>>,
```

Comment:
Is this OK? I noticed that all other JS rules use `Box<[Box<str>]>` for str sequences, but a set sounds more reasonable to me semantically and has a nice side effect of setting `"uniqueItems": true` in config schema.

Reply:
It's probably fine, but it does imply a little overhead when building the hash set initially. Does the `uniqueItems` option actually get rendered out? I think that's the more valuable thing.

Reply:
It does! You can see that option rendered in the schema, it is updated in this PR:) And `uniqueItems` has user-visible impact, at least my editor plugin (LSP-json in Sublime Text) highlights duplicates with such definition:

![image](https://github.com/user-attachments/assets/5b4cba34-f3ef-421a-ba12-9129de9c0a35)